core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -273,8 +276,19 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
 
     override def processNotification(notificationMessage: String) {
       val resource: Resource = Resource.fromString(notificationMessage)
-      val acls = getAclsFromZk(resource)
-      updateCache(resource, acls)
+      try {
+        val acls = getAclsFromZk(resource)
+        updateCache(resource, acls)
+      } catch {
+        case e: InterruptedException =>
+          // It is OK to have InterruptedException after authorizer is closed.
+          if (!isClosed.get)
+            throw e
+          authorizerLogger.debug(s"NotificationHandler is interrupted after authorizer is closed")
```

Comment:
authorizerLogger is used to log whether a request is allowed or denied. So, we shouldn't log other info there. Also, not sure if we need to log this. Finally, it seems the try/catch logic should be added in ZkNodeChangeNotificationListener.processNotifications() since that method has code that reads from ZK directly. Perhaps we can add a close() method in ZkNodeChangeNotificationListener to set the closing indicator.